### PR TITLE
Create patient coordinator

### PR DESCRIPTION
**Pull request by the-kenneth**
# Description

Adds a patient coordinator to help manage the navigation and props in the patient flow. 

There is now a warning that the singletons have a require cycle, it works fine in this case. However this can be sorted by next adding in a parent coordinator and injecting dependencies. But this works for now.

### CODE REVIEW

File: src/CovidApp.tsx [hunk 1]
```diff
@@ -38,7 +38,6 @@ import { ResetPasswordConfirmScreen } from '@covid/features/password-reset/Reset
 import { ResetPasswordScreen } from '@covid/features/password-reset/ResetPassword';
 import AboutYouScreen from '@covid/features/patient/AboutYouScreen';
 import PreviousExposureScreen from '@covid/features/patient/PreviousExposure';
-import StartPatientScreen from '@covid/features/patient/StartPatient';
```
Comment (selvach) [inline src/CovidApp.tsx:41]: NICE! Bye by StartPatientScreen!

File: src/core/patient/PatientCoordinator.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import { StackNavigationProp } from '@react-navigation/stack';
+
+import { PatientStateType } from '@covid/core/patient/PatientState';
+import UserService from '@covid/core/user/UserService';
+import assessmentCoordinator from '@covid/core/assessment/AssessmentCoordinator';
+import { assessmentService } from '@covid/Services';
+
+import { ScreenParamList } from '../../features/ScreenParamList';
```
Comment (selvach) [inline src/core/patient/PatientCoordinator.ts:8]: Use relative import here.

File: src/core/patient/PatientCoordinator.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    this.userService = userService;
+  };
+
+  startPatient = () => {
+    const { currentPatient } = this.patientData;
+    const config = this.userService.getConfig();
+    const patientId = currentPatient.patientId;
+
+    const startPage = 'WelcomeRepeat';
+    const shouldAskStudy = config.enableCohorts && currentPatient.shouldAskStudy;
```
Comment (selvach) [inline src/core/patient/PatientCoordinator.ts:56]: Improved readability! +1

**Comment by the-kenneth** [comment]
Had to make some changes in order to get the tests running. I am now passing the navigator (renamed to AppCoordinator) as a dependancy. This makes it so the child coordinators don't require each other. The AppCoordinator needs to be properly refactored though.

**Comment by abubuwe** [comment]
Random comment - I don't think the Coordinators should live underneath the features folder. Arguably need their own space :)

**Pull request merged**